### ISSUE TITLE
Enthusiastic programmers.look at this!

### ISSUE THREAD

**Issue by vpaintanimation** [pencil2d/pencil#654]
https://github.com/hidefuku/AnimeEffects

**Comment by scribblemaniac** [comment]
If there are some features in that program which you would like to see in Pencil2d, then please structure your post as a feature request for specific features. As it stands right now, this is basically just an advertisement for this alternative project which is not at all what issues are used for.

**Comment by vpaintanimation** [comment]
Sorry, I would like to express that pencil2d has a very good painting operation experience, whether it will join a similar grid deformation animation tool, I see the same GPL license, which means that pencil2d can use its code. I would like to know if pencil2d will integrate such a function.

**Comment by CandyFace** [comment]
While I can't speak for all of us, I will say that integrating such a function is not something i'm interested in before we have fixed our own tools. Right now all focus is on bug fixing, feature requests can come later.

When that's said though, having looked at the video, it looks pretty powerful and deformation like could indeed benefit the animator in some situations. It does not have much to do with traditional 2D animation though, so i'm not sure whether the rest of the team would like to see something like this in Pencil2D.

**Comment by J5lx** [comment]
@vpaintanimation In regard to license compatibility I’d like to point out that Pencil uses GPL2 [which is incompatible to the GPL3](https://opensource.stackexchange.com/a/1778) used by AnimeEffects. Pencil won’t be able to incorporate any AE code unless it switches to GPL3 ([which can be rather non-trivial](https://opensource.stackexchange.com/a/1776)).

**Comment by vpaintanimation** [comment]
Thank you for telling me this, and I'm looking forward to the next of version pencil2d released

**Comment by Jose-Moreno** [comment]
@scribblemaniac @CandyFace @J5lx I agree completely that we have to fix Pencil2D's own tools first. However this tool does provide solution to a rather difficult dilemma I've been posed by colleagues several times, and it is the usability and applicability of Pencil2D in a production environment. 

Hand drawn animation productions are rather scarce these days, with the exception of some video games (http://cupheadgame.com/), tv shows, short films and even some feature films spread around the world, unless you're in Asia (with key players like Japan, Korea and China, which would be a welcoming market for drawn 2D). Mostly anything else is 3D CGI or Cut-Out animation (key players are Canada and the U.S for the latter)

This idea of allowing 2D cut out animation in Pencil2D has been proposed in the past, not to replace the software main principle, but to complement it, considering if you have "tweening" capabilities you can produce faster and mix both animation techniques on the fly (e.g. Layout done with cut-out, final animation done hand drawn)

Now, Pencil2D official vision ever since I can remember Is meant to be a lightweight and simple to use animation tool to realize your work, so basically it's underlying philosophy is that you can open the app, sit down and start animating without the hassle of configuring multiple settings and putting down a line during the first 30 seconds. If we manage to comply with this idea, I don't see reason to object potential tools to assist in the creation of 2D animation. First things first though, and fixing bugs come first as Oliver has stated.

p.s. I feel the whole GPLv2 & v3 debacle is rather cumbersome, but I think it's too early to worry about that.

**Comment by Qwerp-Derp** [comment]
Shouldn't the title be renamed to something that's more relevant to the issue at hand @vpaintanimation? The title seems just clickbait-y.

**Comment by vpaintanimation** [comment]
@Qwerp-Derp I have changed the title, it should be consistent with the content. Thanks for reminding.

**Comment by chchwy** [comment]
Happy to see that if someone wants to implement it ;)

**Comment by Jose-Moreno** [comment]
[2020] After some thought I've decided to leave this explicitly to track a mesh warp functionality meant for manipulating a custom grid of points to provide distortion to a flat surface.

This helps a lot when creating artificial perspective in a drawing, and it is widely used in early stages of animation sketching or for correction purposes.

This could be implemented as a "mode" for the Move tool similar to how Krita has a "mode" for their Transform tool.
